pkg/processor/constraints/constraint.go

```diff
@@ -0,0 +1,46 @@
+package constraints
+
+import (
+	"github.com/arttor/helmify/pkg/helmify"
+	yamlformat "github.com/arttor/helmify/pkg/yaml"
+)
+
+const tolerations = "tolerations"
+const topology = "topologySpreadConstraints"
+const nodeSelector = "nodeSelector"
+
+const topologyExpression = "\n{{- if .Values.topologySpreadConstraints }}\n" +
+	"      topologySpreadConstraints: {{- include \"tplvalues.render\" (dict \"value\" .Values.topologySpreadConstraints \"context\" $) | nindent 8 }}\n" +
+	"{{- end }}\n"
+
+const nodeSelectorExpression = "{{- if .Values.nodeSelector }}\n" +
+	"      nodeSelector: {{- include \"tplvalues.render\" ( dict \"value\" .Values.nodeSelector \"context\" $) | nindent 8 }}\n" +
+	"{{- end }}\n"
+
+const tolerationsExpression = "{{- if .Values.tolerations }}\n" +
+	"      tolerations: {{- include \"tplvalues.render\" (dict \"value\" .Values.tolerations \"context\" .) | nindent 8 }}\n" +
+	"{{- end }}\n"
+
+// ProcessSpecMap adds 'topologyConstraints' to the podSpec in specMap, if it doesn't
+// already has one defined.
+func ProcessSpecMap(name string, specMap map[string]interface{}, values *helmify.Values) string {
+
+	mapConstraint(name, specMap, topology, []interface{}{}, values)
+	mapConstraint(name, specMap, tolerations, []interface{}{}, values)
+	mapConstraint(name, specMap, nodeSelector, map[string]string{}, values)
+
+	spec, err := yamlformat.Marshal(specMap, 6)
+	if err != nil {
+		return ""
+	}
+	return spec + topologyExpression + nodeSelectorExpression + tolerationsExpression
+}
+
+func mapConstraint(name string, specMap map[string]interface{}, constraint string, override interface{}, values *helmify.Values) {
+	if specMap[constraint] != nil {
+		(*values)[name].(map[string]interface{})[constraint] = specMap[constraint].(interface{})
+	} else {
+		(*values)[name].(map[string]interface{})[constraint] = override
```

Comment:
helmify should not add anything new to a template by default. it breaks backward compatibility and may be unexpected. lets template spec only if it is already presented in source manifest.

Reply:
@arttor about that, would it make sense to add these defaults fields behind feature flags? E.g. there is already a fullNameOverride value in the template, but this is not added to the values. Would it make sense to you to be able to select whether to generate such data?

Reply:
i'd propose not adding extra spec to the template if the spec was not presented in the source manifest. In this case we can just remove the else branch.

Reply:
However, this is quite common for a helm chart to have configurable constraints, by default anyway none is there but if a user has stricter requirements they can use the chart out of the box.

Reply:
i agree. I suggest adding a parameter to a helm chart only if it is presented in the original manifest. in current aproach helmify users will get `nodeSelector`,  `tolerations` and `topologySpreadConstraints` in values.yaml out of nowhere after upgrading to the latest version.

Reply:
What if we put such things, that were not in the initial manifests behind a CLI flag so that users can enable them if they need it?

Reply:
that will work!

Reply:
@mowies and @arttor I added a flag to generate default values as you suggested, this way we could add more things later on and the default behaviour stays the same